lib/Sema/CSDiagnostics.cpp

```diff
@@ -2352,3 +2352,133 @@ bool InaccessibleMemberFailure::diagnoseAsError() {
   emitDiagnostic(Member, diag::decl_declared_here, Member->getFullName());
   return true;
 }
+
+bool ImplicitCoercionToAnyFailure::diagnoseAsError() {
+  auto E = getAnchor();
+  auto subExpr = E->getValueProvidingExpr();
+  auto &TC = getTypeChecker();
+  Decl *decl = nullptr;
+
+  if (auto *UDE = dyn_cast<UnresolvedDotExpr>(E)) {
+    auto name = UDE->getName();
+    auto parent = getDC()->getParentForLookup();
+    SmallVector<ValueDecl *, 4> lookupResults;
+    parent->lookupQualified(FromType->lookThroughAllOptionalTypes(), name,
+                            NL_QualifiedDefault, nullptr, lookupResults);
+    if (!lookupResults.empty()) {
+      decl = lookupResults.front();
+    }
+  }
+
+  auto hasImplicitlyUnwrappedResult = [&](Expr *E) -> bool {
```

Comment:
This information has to be looked up in the constraint system by locator, subscripts and other expressions wouldn't have overload choices  set until `ExprRewritter` runs which is not going to happen when problems are diagnosed via fixes.

Reply:
Done!